### DISCUSSION TITLE
How to make opam recognize a manually installed OCaml version

### DISCUSSION THREAD

**Discussion by anzhi0708** [ocaml/opam#5482] [Q&A]
Hello!

A few days ago I installed `OCaml 4.0.*` & `OPAM 2.0.*` - both are very **outdated** versions - from some repo, then installed `OCaml 4.14.1`, which is a newer version, using OPAM.

Then I decided to give **OCaml 5** a try, 

so I uninstalled both OCaml 4.0.* and OPAM, but seemed like OCaml 4.14.1 was not removed from my device. It was still under `~/.opam`; it was installed when I `cd`ed into there and typed `opam switch create 4.14.1`.

Now, I compiled OCaml 5.1.0 from source, manually, on Android, thanks to the good folks [here](https://github.com/ocaml/ocaml/issues/12101), and I reinstalled OPAM using `curl` after that. 

So, right now, there are two versions of OCaml, `4.14.1` and `5.1.0+dev1`.

But OPAM only recognized 4.14.1, I have to type "$PREFIX/bin/ocaml" to run 5.1.0 like this

```
$ $PREFIX/bin/ocaml --version
The OCaml toplevel, version 5.1.0+dev1-2022-06-09
```

which is annoying, how can I configure OPAM properly?

Thanks!

***

* Edit:

```
~/.opam $ eval $(opam env)
~/.opam $ opam switch list
#  switch  compiler
          description
→  4.14.1  ocaml-base-compiler.4.14.1
          4.14.1
~/.opam $
~/.opam $ ocaml --version
The OCaml toplevel, version 4.14.1
~/.opam $ $PREFIX/bin/ocaml --version
The OCaml toplevel, version 5.1.0+dev1-2022-06-09
~/.opam $ which ocaml
/data/data/com.termux/files/home/.opam/4.14.1/bin/ocaml
~/.opam $ echo $PATH
/data/data/com.termux/files/home/.opam/4.14.1/bin:/data/data/com.termux/files/home/apps:/data/data/com.termux/files/home/.opam/4.14.1/bin:/data/data/com.termux/files/home/apps:/data/data/com.termux/files/usr/bin

~/.opam/opam-init $ opam switch 5.1.0+dev
[ERROR] No switch 5.1.0+dev is currently
        installed. Did you mean 'opam switch
        create 5.1.0+dev'?
        Installed switches are:
          - 4.14.1
```

**Comment by kit-ty-kate** [comment]
Did you run `opam switch <name you used for your 5.1 switch>` ? If so, if that didn't change the environment automatically (probably because you didn't accept opam to change your bashrc file during `opam init`), run `eval $(opam env)` in the terminal you want to change the environment in.

**Reply by anzhi0708** [reply]
That 5.1 version was installed from source code, not through OPAM, so I guess that's why OPAM didn't know there's another version there.

```
~/.opam $ eval $(opam env)
~/.opam $ opam switch list
#  switch  compiler
          description
→  4.14.1  ocaml-base-compiler.4.14.1
          4.14.1
~/.opam $
~/.opam $ ocaml --version
The OCaml toplevel, version 4.14.1
~/.opam $ $PREFIX/bin/ocaml --version
The OCaml toplevel, version 5.1.0+dev1-2022-06-09
~/.opam $ which ocaml
/data/data/com.termux/files/home/.opam/4.14.1/bin/ocaml
```

**Comment by anzhi0708** [comment]
I did not give enough information in this discussion; I'm sorry, and I'll close this discussion (since I'm don't know how to delete it) and start a new one with more detailed information.

**Comment by kit-ty-kate** [comment]
no no don't worry, i was about to reply. I think i understand your issue but I'm currently in a meeting

**Comment by kit-ty-kate** [comment]
Ok I'll answer in #5483 I guess.
